frontend/src/metabase/setup/selectors.ts

```diff
@@ -72,3 +76,33 @@ export const getAvailableLocales = (state: State): LocaleData[] => {
 export const getIsEmailConfigured = (state: State): boolean => {
   return getSetting(state, "email-configured?");
 };
+
+export const getSteps = (state: State) => {
+  const usageReason = getUsageReason(state);
+  const activeStep = getStep(state);
+
+  const steps: { key: SetupStep; isActiveStep: boolean }[] = [
+    { key: "welcome" as const },
+    { key: "language" as const },
+    { key: "user_info" as const },
+    { key: "usage_question" as const },
+    usageReason !== ("embedding" as const) && {
+      key: "db_connection" as const,
+    },
+    { key: "data_usage" as const },
+    { key: "completed" as const },
+  ]
+    .filter(isNotFalsy)
```

Comment:
Unfortunately Boolean is not typed correctly, when passed to filter, the returned array will still be typed as possibly containing false which is super annoying, I usually end up writing a util like isNotFalsy in every project 😢 

<img width="1240" alt="image" src="https://github.com/metabase/metabase/assets/1914270/071fa2a2-6f70-48c2-a073-9bc9b120f13b">

Reply:
Good grief! I didn't know that didn't work on TypeScript. I feel you since I've just dealt with typing `Object.entries` too. Sometimes it feels like we need to jump a bunch of hoops to get TypeScript to understand some very normal syntaxs.

Reply:
Oh, my mind was 🤯 when I first read: https://www.mattstobbs.com/object-keys-typescript/
I believe the same thing applies to Object.entries, still I see no danger in making .filter(Boolean) return an array of non falsy values 🤷 , it's actually done in https://github.com/total-typescript/ts-reset

Reply:
Yeah, I get that by design. But we don't like mutating or initializing objects with more keys than the type has described anyway. So most of our use cases shouldn't have to deal with such legwork.

I've used `key as MyKeyType` in the end https://github.com/metabase/metabase/blob/90a63a77028856c8e4f328c8b2d3dc304af710f3/frontend/src/metabase/public/lib/code-templates.ts#L26

but I could imagine doing this every time would be a bit cumbersome.

btw. Let's get that `ts-reset` in. I think it's be a great UX for us.